src/DependencyInjection/KnpUOAuth2ClientExtension.php

```diff
@@ -214,6 +214,8 @@ private function configureProviderAndClient(ContainerBuilder $container, $provid
             $clientDefinition->addMethodCall('setAsStateless');
         }
 
+        $container->setAlias($clientClass, $clientServiceKey);
```

Comment:
This looks great! Two minor things:

```php
$container->setAlias($clientClass, new Alias($clientServiceKey, false));
```

This will create a private alias... no reason for the user to need to fetch this service directly from the container with that id.

Second. There's one edge case: if someone (for some reason) configures *two* clients with the same provider (e.g. 2 facebook clients). In that case, the alias would be created twice, the second would override the first. It would be better to not create *any* alias in this case, so that the user receives an exception if they try to autowire it (it will be up to them to create an alias if they want to autowire).

How to do this easily? Let's add a new private property to the class: `private $duplicateProviderTypes = array()`. Then, we can do something like:

```php
// add an alias, but only if a provider type is used only 1 time
if (!in_array($providerType, $this->duplicateProviderTypes)) {
    // alias already exists? This is a duplicate type, record it
    if ($container->getAlias($clientClass)) {
        $this->duplicateProviderTypes[] = $providerType;
    } else {
        // all good, add the alias
        $container->setAlias($clientClass, new Alias($clientServiceKey, false));
    }
}
```